### PR TITLE
Ci fix

### PR DESCRIPTION
**Pull request by gforsyth**
Moves flake8 checks to circle, stops allowing failures on travis nightly and removes coverage checks from travis since they weren't working anyway.  

I'm happy to have coverage reports back if someone wants to put in the time to get them working properly but otherwise it's just noise.

### CODE REVIEW

**Comment by gforsyth** [comment]
Ok -- this is working correctly now.  Note that the failures on Circle CI are the flake8 failures that should be fixed in #1810 and #1812

**Comment by scopatz** [comment]
Looks like this is failing in some places @gforsyth. Thanks for putting this in!

**Comment by scopatz** [comment]
Thanks @gforsyth! Just the one small issue and this has some failure on CI. Not sure if those are my fault, though...

**Comment by gforsyth** [comment]
Ok, this is ready for another look.  Allows failures on Python nightly since that's now 3.7 -- we could add a 3.6 check somewhere if we felt like it.  flake8 checks done on 3.5 on circle CI

**Comment by astronouth7303** [comment]
I feel like we should have a 3.6 check, so that we know if there's problems with the upcoming version. Assuming any of the CIs support it.

**Comment by gforsyth** [comment]
Travis has a "3.6-dev" option.  I'm happy to add that, although we might need to allow failures for the moment.  I'd like to get this in sooner, so that we're properly checking other PRs

File: requirements-tests.txt [hunk 1]
```diff
@@ -10,5 +10,3 @@ pygments==2.1.3
 codecov==2.0.5
 flake8==2.6.2
 coverage==4.2
-numpy==1.10.4
-matplotlib==1.5.1
```
Comment (astronouth7303) [inline requirements-tests.txt]: Are we not testing xontribs?

Comment (gforsyth) [inline requirements-tests.txt]: I don't think it's worth it on nightly or 3.6 dev. The install time is ugly on travis

**Comment by scopatz** [comment]
Thanks @gforsyth!

**Pull request merged**